src/wallet/wallet_test.go

```diff
@@ -670,6 +672,61 @@ func TestLoadWallet(t *testing.T) {
 	}
 }
 
+func TestLoadDupWallets(t *testing.T) {
+	wlts := Wallets{}
+	dupWlt := &Wallet{}
+
+	for i := 0; i < 3; i++ {
+		var entries []Entry
+		var dupEntries []Entry
+
+		if i%2 != 0 {
+			entry := Entry{
+				Address: cipher.MustDecodeBase58Address("2iNNt6fm9LszSWe51693BeyNUKX34pPaLx8"),
+				Secret:  cipher.NewSecKey(make([]byte, 32)),
+				Public:  cipher.NewPubKey(make([]byte, 33)),
+			}
+
+			dupEntries = append(dupEntries, entry)
+			entries = append(entries, entry)
+
+			dupWltItem := &Wallet{
+				Meta:    map[string]string{},
+				Entries: dupEntries,
+			}
+
+			dupWltItem.setFilename("dup_wlt_" + strconv.Itoa(i) + ".wlt")
+
+			log.Info("Adding Duplicate wlt: ", dupWltItem.Entries[0].Address.String())
+			dupWlt = dupWltItem
+		}
+
+		entries = append(entries,
+			Entry{
+				Address: testutil.MakeAddress(),
+				Secret:  cipher.NewSecKey(make([]byte, 32)),
+				Public:  cipher.NewPubKey(make([]byte, 33)),
+			})
+
+		wltItem := &Wallet{
+			Meta:    map[string]string{},
+			Entries: entries,
+		}
+
+		wltItem.setFilename("wlt_" + strconv.Itoa(i) + ".wlt")
+
+		isLoaded, _ := wlts.isWalletLoaded(wltItem)
+		require.False(t, isLoaded)
+
+		wlts[strconv.Itoa(i)] = wltItem
+	}
+
+	isLoaded, fileName := wlts.isWalletLoaded(dupWlt)
+
+	require.True(t, isLoaded)
+	log.Infof("Wallet %v, has been loaded.", fileName)
+}
```

Comment:
I think you should look at other tests and follow the TDD convention

Reply:
Im not sure, what you mean by that. The test creates a scenario where five wallets are being loaded - two of them are equal.
As far as I know, TDD means that you first implement a test and then you implement your logic based on the test. 

I'll remove that test, as the functionality is already being tested in service_test.go/TestNewServiceDuplicateWlts